pkg/queue/bounded_queue_test.go

```diff
@@ -112,6 +112,80 @@ func TestBoundedQueue(t *testing.T) {
 	assert.False(t, q.Produce("x"), "cannot push to closed queue")
 }
 
+func TestBoundedQueueWithFactory(t *testing.T) {
```

Comment:
is this test essentially a duplicate of the previous one? It's a lot of duplication and complexity. I would rather refactor it into a helper function that can be used with both factory-based and callback-based initialization.